pinot/generative/torch_gvae/model.py

```diff
@@ -6,68 +6,29 @@
 
 
 class GCNModelVAE(nn.Module):
-    """Graph convolutional neural networks for VAE
-    """
     def __init__(self, input_feat_dim, hidden_dim1, hidden_dim2, dropout):
-        """ Construct a VAE with GCN
-        """
         super(GCNModelVAE, self).__init__()
         self.gc1 = GraphConvolution(input_feat_dim, hidden_dim1, dropout, act=F.relu)
-        self.dc = InnerProductDecoder(dropout, act=lambda x: x) # Decoder
-        self.output_regression = torch.nn.ModuleList([
-            GraphConvolution(hidden_dim1, hidden_dim2, dropout, act=lambda x: x),
-            GraphConvolution(hidden_dim1, hidden_dim2, dropout, act=lambda x: x),
-        ])
+        self.gc2 = GraphConvolution(hidden_dim1, hidden_dim2, dropout, act=lambda x: x)
+        self.gc3 = GraphConvolution(hidden_dim1, hidden_dim2, dropout, act=lambda x: x)
+        self.dc = InnerProductDecoder(dropout, act=lambda x: x)
+
+    def parameterization(self, x, adj):
```

Comment:
I made this typo before. This should spell 'parametrization' without the extra e.

Reply:
This is really odd, this file is outdated. Now sure why it pops up in here. The GCNModelVAE should not even have this parametrization function
```
class GCNModelVAE(nn.Module):
    """Graph convolutional neural networks for VAE
    """
    def __init__(self, input_feat_dim, hidden_dim1=32, \
            hidden_dim2=32, hidden_dim3=16, dropout=0.1, \
            log_lik_scale=1):
        """ Construct a VAE with GCN
        """
        super(GCNModelVAE, self).__init__()
        self.linear = nn.Linear(input_feat_dim, hidden_dim1)
        self.gc1 = GN(hidden_dim1, hidden_dim2)

        # Mapping from "latent graph" to predictive distribution parameter
        self.output_regression = nn.ModuleList([
            GN(hidden_dim2, hidden_dim3),
            GN(hidden_dim2, hidden_dim3),
        ])
        # Decoder
        self.dc = InnerProductDecoder(dropout)
        # Relative weight between the KL divergence and the log likelihood term
        self.log_lik_scale = log_lik_scale

    def forward(self, g):
        """ Compute the parameters of the approximate Gaussian posterior
         distribution
        
        Args:
            x (FloatTensor): node features
                Shape (N, D) where N is the number of nodes in the graph
            adj (FloatTensor): adjacency matrix
                Shape (N, N)
        Returns:
            z: (FloatTensor): the latent encodings of the nodes
                Shape (N, hidden_dim2)
        """
        z1 = self.linear(g.ndata["h"])
        z = self.gc1(g, z1)
        return z

    def condition(self, g):
        """ Compute the approximate Normal posterior distribution q(z|x, adj)
        and associated parameters
        """
        z = self.forward(g)
        theta = [parameter.forward(g, z) for parameter in self.output_regression]
        mu  = theta[0]
        logvar = theta[1]
        distribution = torch.distributions.normal.Normal(
                    loc=theta[0],
                    scale=torch.exp(theta[1]))

        return distribution, theta[0], theta[1]
```